machine_learning/benchmarks/base_regressor_benchmark.py

```diff
@@ -0,0 +1,138 @@
+from qiskit.circuit import Parameter
+from qiskit.circuit.library import ZZFeatureMap, RealAmplitudes
+from qiskit.utils import QuantumInstance, algorithm_globals
+from qiskit_machine_learning.algorithms import NeuralNetworkRegressor
+from qiskit_machine_learning.neural_networks import TwoLayerQNN
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import MinMaxScaler
+
+from .datasets import (
+    DATASET_SYNTHETIC_REGRESSION_FEATURES,
+    DATASET_SYNTHETIC_REGRESSION_LABELS,
+    load_ccpp,
+)
+
+DATASET_SYNTHETIC_REGRESSION = "dataset_synthetic_regression"
+DATASET_CCPP_REGRESSION = "dataset_ccpp"
+
+
+class BaseRegressorBenchmark(ABC):
+    """Base class for regressor benchmarks."""
+
+    def __init__(self) -> None:
+
+        quantum_instance_statevector = QuantumInstance(
+            Aer.get_backend("statevector_simulator"),
+            seed_simulator=algorithm_globals.random_seed,
+            seed_transpiler=algorithm_globals.random_seed,
+        )
+        quantum_instance_qasm = QuantumInstance(
+            Aer.get_backend("qasm_simulator"),
+            shots=1024,
+            seed_simulator=algorithm_globals.random_seed,
+            seed_transpiler=algorithm_globals.random_seed,
+        )
+
+        self.backends = {
+            "statevector_simulator": quantum_instance_statevector,
+            "qasm_simulator": quantum_instance_qasm,
+        }
+
+        # prepare synthetic dataset
+        (
+            synth_train_features,
+            synth_test_features,
+            synth_train_labels,
+            synth_test_labels,
+        ) = train_test_split(
+            DATASET_SYNTHETIC_REGRESSION_FEATURES,
+            DATASET_SYNTHETIC_REGRESSION_LABELS,
+            test_size=5,
+            shuffle=False,
+        )
+
+        ccpp_features, ccpp_labels = self._prepare_ccpp()
+        (
+            ccpp_train_features,
+            ccpp_test_features,
+            ccpp_train_labels,
+            ccpp_test_labels,
+        ) = train_test_split(ccpp_features, ccpp_labels, test_size=5, shuffle=False)
+
+        self.datasets = {
+            DATASET_SYNTHETIC_REGRESSION: {
+                "train_features": synth_train_features,
+                "train_labels": synth_train_labels,
+                "test_features": synth_test_features,
+                "test_labels": synth_test_labels,
+            },
+            DATASET_CCPP_REGRESSION: {
+                "train_features": ccpp_train_features,
+                "train_labels": ccpp_train_labels,
+                "test_features": ccpp_test_features,
+                "test_labels": ccpp_test_labels,
+            },
+        }
+
+    def _prepare_ccpp(self) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Prepare the CCPP dataset, we can afford only a tiny subset of the dataset for training.
+        """
+        ccpp_features, ccpp_labels = load_ccpp()
+        ccpp_features = ccpp_features[:25]
+        ccpp_labels = ccpp_labels[:25]
```

Comment:
```suggestion
        num_features = 25
        ccpp_features = ccpp_features[:num_features]
        ccpp_labels = ccpp_labels[:num_features]
```

Reply:
replaced `num_features` to `num_samples`.